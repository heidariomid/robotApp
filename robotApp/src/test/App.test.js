import React from 'react';
import configureStore from 'redux-mock-store';
import {shallow} from 'enzyme';
import App from '../containers/App';

describe('<App />', () => {
	let middlewares;
	let wrapper;
	let mockStore;
	let initialState;
	let store;
	beforeEach(() => {
		middlewares = [];
		mockStore = configureStore(middlewares);
		initialState = {onRequestRobots: jest.fn(), searchField: '', robots: [], isPending: false};
		store = mockStore(initialState);
		wrapper = shallow(<App store={store} />);
	});

	it('renders components', () => {
		expect(wrapper.debug()).toMatchSnapshot();
	});
	it('filteredRobots', () => {
		expect(wrapper.instance().filteredRobots([])).toEqual([]);
	});
});
