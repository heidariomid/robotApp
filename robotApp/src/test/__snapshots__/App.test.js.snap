// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders components 1`] = `"<App store={{...}} onSearchChange={[Function: onSearchChange]} onRequestRobots={[Function: onRequestRobots]} storeSubscription={{...}} />"`;
